src/compute/src/logging/compute.rs

```diff
@@ -780,165 +780,167 @@ impl<A: Allocate> DemuxHandler<'_, '_, A> {
-                operator,
-                address,
-            }) => self.handle_arrangement_heap_size_operator(operator, address),
-            ComputeEvent::ArrangementHeapSizeOperatorDrop(ArrangementHeapSizeOperatorDrop {
-                operator,
-            }) => self.handle_arrangement_heap_size_operator_dropped(operator),
-            ComputeEvent::DataflowShutdown(DataflowShutdown { dataflow_index }) => {
-                self.handle_dataflow_shutdown(dataflow_index)
+            ComputeEvent::ArrangementHeapSize(inner) => self.handle_arrangement_heap_size(inner),
+            ComputeEvent::ArrangementHeapCapacity(inner) => {
+                self.handle_arrangement_heap_capacity(inner)
             }
-            ComputeEvent::ErrorCount(ErrorCount { export_id, diff }) => {
-                self.handle_error_count(export_id, diff)
+            ComputeEvent::ArrangementHeapAllocations(inner) => {
+                self.handle_arrangement_heap_allocations(inner)
             }
-            ComputeEvent::Hydration(Hydration { export_id }) => self.handle_hydration(export_id),
-            ComputeEvent::LirMapping(LirMapping { global_id, mapping }) => {
-                self.handle_lir_mapping(global_id, mapping)
+            ComputeEvent::ArrangementHeapSizeOperator(inner) => {
+                self.handle_arrangement_heap_size_operator(inner)
             }
-            ComputeEvent::DataflowGlobal(DataflowGlobal { id, global_id }) => {
-                self.handle_dataflow_global(id, global_id)
+            ComputeEvent::ArrangementHeapSizeOperatorDrop(inner) => {
+                self.handle_arrangement_heap_size_operator_dropped(inner)
             }
+            ComputeEvent::DataflowShutdown(shutdown) => self.handle_dataflow_shutdown(shutdown),
+            ComputeEvent::ErrorCount(error_count) => self.handle_error_count(error_count),
+            ComputeEvent::Hydration(hydration) => self.handle_hydration(hydration),
+            ComputeEvent::LirMapping(mapping) => self.handle_lir_mapping(mapping),
+            ComputeEvent::DataflowGlobal(global) => self.handle_dataflow_global(global),
         }
     }
 
-    fn handle_export(&mut self, id: GlobalId, dataflow_id: usize) {
+    fn handle_export(
+        &mut self,
+        Export {
+            export_id,
+            dataflow_index,
+        }: Export,
+    ) {
         let ts = self.ts();
-        let datum = ExportDatum { id, dataflow_id };
+        let datum = ExportDatum {
+            export_id,
+            dataflow_index,
+        };
         self.output.export.give((datum, ts, 1));
 
-        let existing = self.state.exports.insert(id, ExportState::new(dataflow_id));
+        let existing = self
+            .state
+            .exports
+            .insert(export_id, ExportState::new(dataflow_index));
         if existing.is_some() {
-            error!(export = %id, "export already registered");
+            error!(export = %export_id, "export already registered");
```

Comment:
Nit: You could simplify this log (and all the ones below) to not have an extra field name, i.e. `error!(%export_id, "export already registered")`. Having the extra name avoided confusion when the variable was just a generic `id`, but now the variable name is self-describing.

Reply:
Ack, done.